cdci_data_analysis/analysis/ontology.py

```diff
@@ -0,0 +1,364 @@
+import rdflib as rdf
+from rdflib.collection import Collection
+from rdflib.namespace import RDF, RDFS, OWL, XSD
+import logging
+from cdci_data_analysis.analysis.exceptions import RequestNotUnderstood
+import builtins
+from copy import deepcopy
+
+logger = logging.getLogger(__name__)
+
+ODA = rdf.Namespace("http://odahub.io/ontology#")
+a = RDF.type
+
+def xsd_type_to_python_type(xsd_uri):
+    # TODO: this works only with simple builtin types, but OK for now
+    typename = str(xsd_uri).split('#')[-1]
+    if typename == 'integer': typename = 'int'
+    if typename == 'boolean': typename = 'bool'
+    if typename == 'string': typename = 'str'
+    try:
+        return getattr(builtins, typename)
+    except AttributeError:
+        return None
+
+class MainOntologyGraph:
+    def __init__(self, ontology_path, version):
+        self._ver = version 
+        self._path = ontology_path
+        self._g = rdf.Graph()
+        self._g.parse(ontology_path)
+        self._g.bind('oda', ODA)
+        
+    @property
+    def ontology_path(self):
+        return self._path
+    
+    @property 
+    def version(self):
+        return self._ver
+    
+    @property
+    def graph(self):
+        return deepcopy(self._g)
+    
+    def reset(self, ontology_path, version):
+        if version != self._ver or ontology_path != self._path:
+            self._g = rdf.Graph()
+            self._g.parse(ontology_path)
+            self._g.bind('oda', ODA)
```

Comment:
I believe we opted to only use `http://` terms everywhere for consistency. But the prefix could be safely added, just in case `https://` is ocasionaly used somehere